### DISCUSSION TITLE
Deno imports no longer working in v 1.4

### DISCUSSION THREAD

**Discussion by andrewpbray** [quarto-dev/quarto-cli#8521] [Q&A] [answered]
### Description

I just upgraded quarto v1.3 to quarto 1.4 and I'm now getting an error when trying to run a .ts script.

```bash
(base) MacBookPro:deno-reprex andrewbray$ quarto run test-script.ts
error: Specifier not found in cache: "https://deno.land/std@0.204.0/encoding/yaml.ts", --cached-only is specified.
    at file:///Users/andrewbray/Desktop/deno-reprex/test-script.ts:1:23
```
This is in a [minimal repo](https://github.com/andrewpbray/deno-reprex) that is the boilerplate website plus a `test-script.ts` file containing:
```typescript
import { parse } from "https://deno.land/std/encoding/yaml.ts";

const config = parse(Deno.readTextFileSync("_quarto.yml"));
```
Which is the example taken from the [quarto docs](https://quarto.org/docs/projects/scripts.html#deno-scripts).

Version info:
```
(base) MacBookPro:deno-reprex andrewbray$ quarto --version
1.4.549
```

**Comment by cscheid** [comment]
Ah, that's annoying. This is a bug on our import maps for user scripts.

**Reply by mcanouil** [reply]
Closing in favour of:
- https://github.com/quarto-dev/quarto-cli/issues/8522

**Comment by cscheid** [comment, marked as answer]
@andrewpbray: Oh, actually we did do it. The issue is that deno changed their stdlib in a way that we can't actually paper over.

You need to import from a different path, unfortunately:

```
import { parse } from "https://deno.land/std/yaml/mod.ts";
```

We don't document these changes because we don't actually know it ourselves. As far as I can tell, there's no change log in deno's stdlib that describes the path or API changes :(

**Reply by andrewpbray** [reply]
Ah, ok, makes sense.

I've put in a PR to update the docs:

[https://github.com/quarto-dev/quarto-web/pull/996](https://github.com/quarto-dev/quarto-web/pull/996)

**Reply by cscheid** [reply]
🤦 Thank you!

(@dragonstyle, @cwickham: this is just a note to highlight this general class of problem we have with docs, where the docs don't include testing that ensures that the docs are correct. I don't know what our solution should be precisely. But the fact that _Quarto_ documentation, of all pieces of software, is prone to this, I think is salient.)
